include/swift/Reflection/ReflectionContext.h

```diff
@@ -644,6 +645,47 @@ class ReflectionContext
     return false;
   }
 
+  /// Adds an image using the FindSection closure to find the swift metadata
+  /// sections. \param[in] FindSection
+  ///     Closure that finds sections by name. ReflectionContext is in charge
+  ///     of freeing the memory buffer in the RemoteRef return value.
+  ///     process.
+  /// \return
+  ///     /b True if any of the reflection sections were registered,
+  ///     /b false otherwise.
+  bool addImage(llvm::function_ref<
+                std::pair<RemoteRef<void>, uint64_t>(ReflectionSectionKind)>
+                    FindSection) {
+    auto Sections = {
+        ReflectionSectionKind::fieldmd, ReflectionSectionKind::assocty,
+        ReflectionSectionKind::builtin, ReflectionSectionKind::capture,
+        ReflectionSectionKind::typeref, ReflectionSectionKind::reflstr};
+
+    llvm::SmallVector<std::pair<RemoteRef<void>, uint64_t>, 6> Pairs;
+    for (auto Section : Sections) {
+      Pairs.push_back(FindSection(Section));
+      auto LatestRemoteRef = std::get<RemoteRef<void>>(Pairs.back());
+      if (LatestRemoteRef) {
+        MemoryReader::ReadBytesResult Buffer(
+            LatestRemoteRef.getLocalBuffer(),
+            [](const void *Ptr) { free(const_cast<void *>(Ptr)); });
+
+        savedBuffers.push_back(std::move(Buffer));
+      }
+    }
+
+    // If we didn't find any sections, return.
+    if (llvm::all_of(Pairs, [](const auto &Pair) { return !Pair.first; }))
+      return false;
+
+    ReflectionInfo Info = {
+        {Pairs[0].first, Pairs[0].second}, {Pairs[1].first, Pairs[1].second},
```

Comment:
Are those pairs, too. Ie.: Would `ReflectionInfo Info = {Pairs[0], Pairs[1], ...}` work, too?

Reply:
That doesn't work, unfortunately :(

Reply:
I looked at this too, they're not pairs, unfortunately. It's too bad we don't have a way to construct from a pair, like `{*pair}` expanding to `{pair.first, pair.second}`.